env-builder/src/flox-env.cc

```diff
@@ -0,0 +1,309 @@
+      auto package_flake
+        = flake::lockFlake( state, package_input_ref, flake::LockFlags {} );
+
+      auto vFlake = state.allocValue();
+      flake::callFlake( state, package_flake, *vFlake );
+      state.forceAttrs( *vFlake, noPos, "while parsing flake" );
+
+
+      auto output = vFlake->attrs->get( state.symbols.create( "outputs" ) );
+
+
+      /* evaluate the package */
+      for ( auto path_segment : package.attrPath )
+        {
+          state.forceAttrs( *output->value,
+                            output->pos,
+                            "while parsing cached flake data" );
+
+          auto next
+            = output->value->attrs->get( state.symbols.create( path_segment ) );
+
+          if ( ! next )
+            {
+              std::ostringstream str;
+              output->value->print( state.symbols, str );
+              throw Error( "Attribute '%s' not found in set '%s'",
+                           path_segment,
+                           str.str() );
+            }
+          output = next;
+        }
+
+
+      auto package_drv = getDerivation( state, *output->value, false );
+
+      if ( ! package_drv.has_value() )
+        {
+          throw Error( "Failed to get derivation for package '%s'",
+                       nlohmann::json( package ).dump().c_str() );
+        }
+
+      /* Collect all outputs to include in the environment */
+      for ( auto output : package_drv->queryOutputs() )
+        {
+          if ( ! output.second.has_value() )
+            {
+              continue;
+            }  // skip outputs without path
+          pkgs.emplace_back(
+            state.store->printStorePath( output.second.value() ),
+            true,
+            package.priority );
+          references.insert( output.second.value() );
+        }
+
+      /* Collect drvs that may yet need to be built */
+      if ( auto drvPath = package_drv->queryDrvPath() )
+        {
+          drvsToBuild.push_back( { *drvPath } );
+        }
+    }
+
+  /* Build derivations that make up the environment */
+  // todo check if this builds `outputsToInstall` only
+  state.store->buildPaths( toDerivedPaths( drvsToBuild ),
+                           state.repair ? bmRepair : bmNormal );
+
+  // todo: is it script _xor_ file?
+  if ( auto hook = lockfile.getManifest().getManifestRaw().hook )
+    {
+      nix::Path script_path;
+
+      if ( auto script = hook->script )
+        {
+          script_path = createTempFile().second;
+          std::ofstream file( script_path );
+          file << script.value();
+          file.close();
+        }
+
+
+      if ( auto file = hook->file ) { script_path = file.value(); }
+
+      if ( ! script_path.empty() )
+        {
+
+          auto tempDir = createTempDir();
+          std::filesystem::copy_file( script_path,
+                                      std::filesystem::path( tempDir )
+                                        / "activation-hook.sh" );
```

Comment:
```suggestion
                                        / "bin" / "activation-hook.sh" );
```

And also make it executable and also create the `bin` folder